AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts

```diff
@@ -104,32 +107,62 @@ export class AssignJudgeComponent extends BookingBaseComponent implements OnInit
     }
 
     private initForm() {
-        const existingJudge = this.hearing.participants.find(x => x.is_judge);
-
-        this.populateFormFields(existingJudge);
+        this.initFormFields();
 
         this.form = this.fb.group({
-            judgeName: [this.judge.email, Validators.required],
             judgeDisplayNameFld: this.judgeDisplayNameFld,
             judgeEmailFld: this.judgeEmailFld,
             judgePhoneFld: this.judgePhoneFld
         });
 
-        this.updateJudgeSummary();
+        this.setFieldSubscription();
     }
 
-    updateJudgeSummary() {
-        this.$subscriptions.push(
-            this.judgeName.valueChanges.subscribe(judgeUserId => {
-                this.addJudge(judgeUserId);
-                this.isJudgeSelected = judgeUserId !== null;
-                if (this.isJudgeSelected) {
-                    this.judgeEmailFld.setValue(this.otherInformationDetails.JudgeEmail);
-                    this.judgePhoneFld.setValue(this.otherInformationDetails.JudgePhone);
-                    this.canNavigate = this.isJudgeSelected;
-                }
-            })
-        );
+    private initFormFields() {
+        this.judgeDisplayNameFld = new FormControl(this.judge?.display_name, {
+            validators: [Validators.required, Validators.pattern(Constants.TextInputPattern), Validators.maxLength(255)],
+            updateOn: 'blur'
+        });
+        this.judgeEmailFld = new FormControl(this.otherInformationDetails.JudgeEmail, {
+            validators: [Validators.pattern(Constants.EmailPattern), Validators.maxLength(255)],
+            updateOn: 'blur'
+        });
+        this.judgePhoneFld = new FormControl(this.otherInformationDetails.JudgePhone, {
+            validators: [Validators.pattern(Constants.PhonePattern)],
+            updateOn: 'blur'
+        });
+    }
+
+    updateJudge(judge: ParticipantModel) {
+        this.judge = judge;
+        this.canNavigate = false;
+        this.isJudgeParticipantError = false;
+        if (judge) {
+            if (this.hearingService.canAddJudge(judge.username)) {
+                judge.is_judge = true;
+                this.courtAccountJudgeEmail = judge.email;
+                this.judgeDisplayNameFld.setValue(this.judge.display_name);
+                this.hearing.participants = this.hearing.participants.filter(x => !x.is_judge);
+                this.hearing.participants.unshift(judge);
+                this.hearing = Object.assign({}, this.hearing);
+                this.canNavigate = true;
+            } else {
+                this.isJudgeParticipantError = true;
+            }
+        } else {
+            this.removeJudge();
+        }
+
+        this.isJudgeSelected = judge != null;
```

Comment:
!!judge maybe? What's this for? Maybe this one could just be a getter as it looks like it's connected to this.judge anyway?

Reply:
Yeah there's a few of these types of things sprinkled around that I was hesitant to change. I think this one can be no problem